sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -656,6 +659,73 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
     }
   }
 
+  /**
+   * Convert cudf unsigned integer to wider signed integer that parquet expects
+   * After spark 3.2.0, parquet read uint8 as int16, uint16 as int32, uint32 as int64
+   *
+   * @param inputTable
+   * @return
+   */
+  private def convertToUnsigned(inputTable: Table) = {
+    var typeCastingNeeded = needUnsignedCast(inputTable, readDataSchema)
+    if (typeCastingNeeded) {
+      val newColumns = new Array[ColumnVector](readDataSchema.length)
+      try {
+        withResource(inputTable) { table =>
+          (0 until readDataSchema.length).foreach(i => {
+            val readField = readDataSchema(i)
+            val origCol = table.getColumn(i)
+            val col: ColumnVector =
+              ColumnCastUtil.ifTrueThenDeepConvertTypeAtoTypeB(origCol, readField.dataType,
+                (dt, cv) => needUnsignedToSignedCastFunction(cv, dt),
```

Comment:
```suggestion
                (dt, cv) => needUnsignedToSignedCast(cv, dt),
``` 
?

Reply:
The function ifTrueThenDeepConvertTypeAtoTypeB will search the pattern in DFS mode, so this needUnsignedToSignedCastFunction is ok.